src/forms/form-fields/drop-down-async.js

```diff
@@ -142,7 +142,8 @@ class DropDownAsync extends Component {
                         )),
                     });
                 }
-            });
+            })
+            .then(() => this.onChange({ target: { value: this.props.value ? this.props.value.id : this.props.value} }))
```

Comment:
`this.props.value ? this.props.value.id : this.props.value`?

Why return the falsey `props.value`? I think it would be more clear to return something specific.

Reply:
To be honest, it's just because the way the value are passed down to the onchange handler to the DropDown. See line 185. 
I will change this either way though.

Reply:
Oh ok, it just looked a bit strange.